### PR TITLE
[2.x] Changed all request and Auth user calls to use the correct config('jetstream.guard') value.

### PR DESCRIPTION
**Pull request by mxm1070**
I'm working on a multitenant application with a separate frontend and backend auth. The frontend uses a customer auth guard and the backend uses the default web auth guard. Everything is configured correctly and works fine as long as I sign into the frontend first and then the backend. If I sign into the backend and then try to load the frontend I get errors with Inertia in the ShareInertiaData middleware, specifically this section:

```php
'user' => function () use ($request) {
    if (! $request->user()) {
        return;
    }

    if (Jetstream::hasTeamFeatures() && $request->user()) {
        $request->user()->currentTeam;
    }

    return array_merge($request->user()->toArray(), array_filter([
        'all_teams' => Jetstream::hasTeamFeatures() ? $request->user()->allTeams()->values() : null,
    ]), [
        'two_factor_enabled' => ! is_null($request->user()->two_factor_secret),
    ]);
},
```

I realized that Inertia is resolving the wrong user if I'm not signed into the frontend first because the $request->user() calls do not respect the config('jetstream.guard'), and I'm using teams on the frontend, but not the backend so ! $request->user() doesn't catch it. 

I fixed this file to respect the config guard and it has fixed my problems. 

```php
<?php

namespace Laravel\Jetstream\Http\Middleware;

use Illuminate\Support\Facades\Gate;
use Illuminate\Support\Facades\Session;
use Inertia\Inertia;
use Laravel\Fortify\Features;
use Laravel\Jetstream\Jetstream;

class ShareInertiaData
{
    /**
     * Handle the incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  callable  $next
     * @return \Illuminate\Http\Response
     */
    public function handle($request, $next)
    {
        Inertia::share(array_filter([
            'jetstream' => function () use ($request) {
                return [
                    'canCreateTeams' => $request->user(config('jetstream.guard')) &&
                                        Jetstream::hasTeamFeatures() &&
                                        Gate::forUser($request->user(config('jetstream.guard')))->check('create', Jetstream::newTeamModel()),
                    'canManageTwoFactorAuthentication' => Features::canManageTwoFactorAuthentication(),
                    'canUpdatePassword' => Features::enabled(Features::updatePasswords()),
                    'canUpdateProfileInformation' => Features::canUpdateProfileInformation(),
                    'flash' => $request->session()->get('flash', []),
                    'hasAccountDeletionFeatures' => Jetstream::hasAccountDeletionFeatures(),
                    'hasApiFeatures' => Jetstream::hasApiFeatures(),
                    'hasTeamFeatures' => Jetstream::hasTeamFeatures(),
                    'hasTermsAndPrivacyPolicyFeature' => Jetstream::hasTermsAndPrivacyPolicyFeature(),
                    'managesProfilePhotos' => Jetstream::managesProfilePhotos(),
                ];
            },
            'user' => function () use ($request) {
                if (! $request->user(config('jetstream.guard'))) {
                    return;
                }

                if (Jetstream::hasTeamFeatures() && $request->user(config('jetstream.guard'))) {
                    $request->user(config('jetstream.guard'))->currentTeam;
                }

                return array_merge($request->user(config('jetstream.guard'))->toArray(), array_filter([
                    'all_teams' => Jetstream::hasTeamFeatures() ? $request->user(config('jetstream.guard'))->allTeams()->values() : null,
                ]), [
                    'two_factor_enabled' => ! is_null($request->user(config('jetstream.guard'))->two_factor_secret),
                ]);
            },
            'errorBags' => function () {
                return collect(optional(Session::get('errors'))->getBags() ?: [])->mapWithKeys(function ($bag, $key) {
                    return [$key => $bag->messages()];
                })->all();
            },
        ]));

        return $next($request);
    }
}

```

But I started looking around and noticed that most, if not all, of the calls to either $request->user() and Auth::user() do not respect the config guard. So it went through and replaced those as well.

In summary:
```php
$request()->user() => $request->user(config('jetstream.guard'))
Auth::user() => Auth::guard(config('jetstream.guard'))->user()
```

### CODE REVIEW

**Comment by taylorotwell** [comment]
It's because once a request is authenticated with a given guard that guard is set to be the default guard.
